doc/examples/contextual_enhancement.py

```diff
@@ -0,0 +1,278 @@
+# -*- coding: utf-8 -*-
+"""
+==============================================
+Contextual enhancement
+==============================================
+
+This demo presents an example of crossing-preserving contextual enhancement of 
+FOD/ODF fields [Meesters2016_ISMRM_], implementing the contextual PDE framework 
+of [Portegies2015_PLoSOne]_ for processing HARDI data. The aim is to enhance the 
+alignment of elongated structures in the data such that crossing/junctions are 
+maintained while reducing noise and small incoherent structures. This is achieved
+ via a hypo-elliptic 2nd order PDE in the domain of coupled positions and 
+ orientations :math:`\mathbb{R}^3 \rtimes S^2`. This domain carries a non-flat 
+ geometrical differential structure that allows including a notion of alignment 
+ between neighboring points. 
+
+Let :math:`({\bf y},{\bf n}) \in \mathbb{R}^3\rtimes S^2` where 
+:math:`{\bf y} \in \mathbb{R}^{3}` denotes the spatial part, and 
+:math:`{\bf n} \in S^2` the angular part. 
+Let :math:`W:\mathbb{R}^3\rtimes S^2\times \mathbb{R}^{+} \to \mathbb{R}` be 
+the function representing the evolution of FOD/ODF field. Then, the contextual 
+PDE with evolution time :math:`t\geq 0` is given by:
+
+.. math::
+
+    \begin{cases}
+	\frac{\partial}{\partial t} W({\bf y},{\bf n},t) &= ((D^{33}({\bf n} \cdot 
+	        \nabla)^2 + D^{44} \Delta_{S^2})W)({\bf y},{\bf n},t)
+	\\ W({\bf y},{\bf n},0) &= U({\bf y},{\bf n})
+	\end{cases},
+
+where:
+
+* :math:`D^{33}>0` is  the coefficient for the spatial smoothing (which goes 
+	only in the direction of :math:`n`);
+
+* :math:`D^{44}>0` is the coefficient for the angular smoothing 
+(here :math:`\Delta_{S^2}` denotes the Laplace-Beltrami operator 
+	on the sphere :math:`S^2`);
+
+* :math:`U:\mathbb{R}^3\rtimes S^2 \to \mathbb{R}` is the initial condition 
+given by the noisy FOD/ODF’s field.
+
+This equation is solved via a shift-twist convolution (denoted by 
+	:math:`\ast_{\mathbb{R}^3\rtimes S^2}`) with its corresponding kernel 
+:math:`P_t:\mathbb{R}^3\rtimes S^2 \to \mathbb{R}^+`:
+
+.. math::
+
+	W(\vec{y},\vec{n},t) = (P_t \ast_{\mathbb{R}^3 \rtimes S^2} U)(\vec{y},\vec{n}) 
+	= \int_{\mathbb{R}^3} \int_{S^2} P_t (R^T_{\vec{n}^\prime}(\vec{y}-\vec{y}^\prime),
+	 R^T_{\vec{n}^\prime} \vec{n} ) U(\vec{y}^\prime, \vec{n}^\prime) 
+\hspace{0.2em} d\sigma (\vec{n}^\prime) d\vec{y}^\prime.
```

Comment:
This line doesn't get properly rendered in sphinx. Maybe you need to not have the linebreak here. It's OK to have lines of more than 80 characters, when there's no other choice!